CRM/Member/Form.php

```diff
@@ -77,6 +77,23 @@ class CRM_Member_Form extends CRM_Contribute_Form_AbstractEditPayment {
   public $_priceSetId;
 
   /**
+   * Used when price set specified.  Used to indicate that form is only to retrieve
+   * price set.  How we use price set is unconventional.  We re-use the same
+   * Membership/MembershipRenewal Form used to enter memberships.  When user selects
+   * a PriceSet, we use JavaScript to inject HTML returned by a recursive call to this form.
+   *
+   * This flag controls this recursive call, short-circuiting it to only retrieve the price
```

Comment:
I was about to comment on this, and then saw your WIP#9387, which is pretty well where my comment was going to head with some extra bits about Angular.

Might be OK if time allows it.  Unfortunately, I'm out of coding cycles, so this could send it for a spin depending on how far you get into this.  Maybe as a separate JIRA?

The words "real price set" also made me think would be nice to have a test case that includes a "real price set" -- the complex one that the CIC uses.